crates/rolldown_binding/src/types/binding_output_chunk.rs

```diff
@@ -3,113 +3,255 @@ use std::collections::HashMap;
 use napi_derive::napi;
 use rolldown_sourcemap::SourceMap;
 
-use crate::types::binding_rendered_module::BindingRenderedModule;
+use crate::{type_aliases::WeakRefMutex, types::binding_rendered_module::BindingRenderedModule};
 
 #[napi]
 pub struct BindingOutputChunk {
-  inner: &'static mut rolldown_common::OutputChunk,
+  inner: WeakRefMutex<Vec<rolldown_common::Output>>,
+  index: usize,
 }
 
 #[napi]
 impl BindingOutputChunk {
-  pub fn new(inner: &'static mut rolldown_common::OutputChunk) -> Self {
-    Self { inner }
+  pub fn new(inner: WeakRefMutex<Vec<rolldown_common::Output>>, index: usize) -> Self {
```

Comment:
The `index` maybe is invalid, becuase maybe `remove` opreation.

Reply:
I don't want to use https://github.com/rolldown/rolldown/pull/1905#discussion_r1709337649 so far. Is it ok to change the code back to the old unsafe behavior?

Reply:
Yeah. Just to change the code back, or using `fileName` to search it is ok .

Reply:
No. Using `fileName` search is good to me, i can't meet found segment fault.